### ISSUE TITLE
Make it clear that certain metrics, when added to a MetricFrame, need the sample_params argument, and how to fill it out.

### ISSUE THREAD

**Issue by tuffstuff9** [fairlearn/fairlearn#1222]
#### Describe the issue linked to the documentation

[Here](https://fairlearn.org/main/user_guide/assessment/advanced_metricframe.html) is the section in the docs which gives an 'example' of how sample_params is meant to be used. It simply gives a description which is not entirely clear, considering it took me over 2 hours of searching around and trying different things until I finally got my code to work.

#### Suggest a potential alternative/fix

Something like this is a lot clearer at illustrating what the sample_params does:
![image](https://user-images.githubusercontent.com/57072903/224558009-721d0031-48e5-4dca-9fc8-74f355757a1c.png)

However, it is still not entirely clear to me why I need to have the 'sensitive_features' included in the sample_params in the first place, when it is already included in the MetricFrame. The errors thrown out when you forgo the sample_params led me down the wrong path entirely, as can be seen below. 

![image](https://user-images.githubusercontent.com/57072903/224558103-ccbc32a0-083c-4a9e-bb76-7211d639bf80.png)

This makes me think that I need to include the sensitive_features argument in my one/all of the metrics in the metric_dict, which then says I need a y_true and a y_pred, and once I include those, it gives me this error:

![image](https://user-images.githubusercontent.com/57072903/224558195-c5eeb381-26fd-4dd7-a0e6-789c6a62d605.png)
![image](https://user-images.githubusercontent.com/57072903/224558959-5601ca4f-fa4c-4ad2-80a5-d54d5678cec9.png)

And at this point, one reaches a dead end and goes back to the drawing board. I think these error messages can be improved, but more importantly, I am not sure why the docs don't clearly specify why the metrics with _ratio or _difference (at the very least these are the ones I know of) need a sample_params to be passed into the MetricFrame, and what the nature of the sample_params must be. 

Here is what the docs have to say about the sample_params parameter:

![image](https://user-images.githubusercontent.com/57072903/224558368-55413c22-b0af-478f-939a-c2c2bad20f1d.png)

Perhaps I am missing something, but I am not sure how my above screenshot of a correct sample_params entails "the values being the string-to-array-like dictionaries". Also, I do not see any distinction in the description of the actual metrics regarding whether they need a sample_params to be passed in or not. For example 'false_positive_rate' does not, but 'equalized_odds_ratio' does. 

Thank you for taking this into consideration.

**Comment by hildeweerts** [comment]
Hi, @tuffstuff9. Thank you for sharing your questions with us - it helps us a lot in improving our documentation.

Fairlearn has two primary ways to assess fairness:

* `fairlearn.metrics.MetricFrame`: a data structure that can be used to perform a disaggregated analysis of a particular base metric (e.g., selection rate, false positive rate, etc.) across sensitive groups. `MetricFrame` lets you quickly assess how the model performs across sensitive groups. Additionally, it has the functionality to perform different types of aggregations on the disaggregated analysis. For example, `MetricFrame.difference()` allows you to compute the maximum difference between groups (e.g., the maximum difference in selection rates across genders).
* A set of predefined fairness metrics that output a single number: e.g., `fairlearn.metrics.false_positive_rate_difference`, `fairlearn.metrics.equalized_odds_difference`. Under the hood, these fairness metrics use `MetricFrame` to compute a particular base metric (e.g., false positive rate) across sensitive groups and subsequently perform a particular aggregation on the base metric values across groups (e.g., the maximum difference between groups in case of `false_positive_rate_difference`).

In other words, `MetricFrame` is designed to take as input **base** metrics such as `accuracy_score`, `false_negative_rate` and `recall`, but **not** predefined **fairness** metrics such as `false_negative_rate_difference` and `demographic_parity_ratio`. This is why you encounter issues with metrics such as `false_negative_rate_difference`: these metrics take as input `sensitive_features` to be able to compute the maximum difference between groups. However, computing the maximum difference in false positive rates across sensitive groups *within* one sensitive group is not very informative.

I hope this helps! Let me know if you have any questions. Also, if you have suggestions on where / how to make this more clear in our documentation, I'd love to hear them!

**Comment by hildeweerts** [comment]
Hi @tuffstuff9. I hope our answer was useful. I will close this issue now. If you have any remaining questions, let us know!
